src/zarr/api/asynchronous.py

```diff
@@ -66,6 +65,31 @@
     "zeros_like",
 ]
 
+# Persistence mode: 'r' means read only (must exist); 'r+' means
+# read/write (must exist); 'a' means read/write (create if doesn't
+# exist); 'w' means create (overwrite if exists); 'w-' means create
+# (fail if exists).
+persistence_to_store_modes: dict[AccessModeLiteral, StoreAccessMode] = {
+    "r": "r",
+    "r+": "w",
+    "a": "w",
+    "w": "w",
+    "w-": "w",
+}
+
+
+def _handle_store_mode(mode: AccessModeLiteral | None) -> StoreAccessMode:
+    if mode is None:
+        return "r"
+    else:
+        return persistence_to_store_modes[mode]
```

Comment:
What do we have to change higher up in the codebase if we narrow the function signature to `mode: AccessModeLiteral`? My gut feeling is that we should not need to handle `None` here, and the `None` -> `r` normalization can happen earlier.